File: src/main/java/com/example/demo/service/CryptoService.java
package com.example.demo.service;

import com.example.demo.util.EncryptionUtils;
import jakarta.annotation.PostConstruct;
import org.springframework.stereotype.Service;

@Service
public class CryptoService {

    private final String rawKey;

    public CryptoService(com.example.demo.config.AppProperties properties) {
        this.rawKey = properties.getEncryption().getKey();
    }

    @PostConstruct
    public void init() {
        EncryptionUtils.setKey(rawKey);
    }

    public String encrypt(String value) {
        return EncryptionUtils.encrypt(value);
    }

    public String decrypt(String value) {
        return EncryptionUtils.decrypt(value);
    }
}

File: src/main/kotlin/com/example/demo/service/CryptoService.kt
package com.example.demo.service

import com.example.demo.util.EncryptionUtils
import jakarta.annotation.PostConstruct
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class CryptoService(
    @Value("\${app.encryption.key:}") private val rawKey: String?
) {
    @PostConstruct
    fun init() {
        EncryptionUtils.setKey(rawKey)
    }

    fun encrypt(value: String?): String? = EncryptionUtils.encrypt(value)

    fun decrypt(value: String?): String? = EncryptionUtils.decrypt(value)
}
